sdk/core/azure-core/inc/azure/core/md5.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utility functions to help compute the MD5 hash value for the input binary data.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Represents the class for the MD5 hash function which maps binary data of an arbitrary
+   * length to small binary data of a fixed length.
+   */
+  class Md5 {
+  public:
+    /**
+     * @brief Construct a default instance of @Md5.
+     */
+    Md5();
+
+    /**
+     * @brief Cleanup any state when destroying the instance of @Md5.
+     */
+    ~Md5();
+
+    /**
+     * @brief Used to append parital binary input data to compute the hash in a streaming fashion.
+     * @param data The pointer to the current block of binary data that is used for hash
+     * calculation.
+     * @param length The size of the data provided.
+     * @remark Once all the data has been added, call #Digest() to get the computed hash value.
+     */
+    void Update(const uint8_t* data, std::size_t length);
+
+    /**
+     * @brief Computes the hash value of all the binary input data appended to the instance so far.
```

Comment:
```suggestion
     * @brief Computes the hash value of all the binary input data appended to the instance.
```

Does it add any value to mention the `so far` ?

Reply:
imo, yes, it let's the user know that we are not only looking at the data provided to this method call but previous Updates too.